Test build failure: The type 'EventPipeProvider' is defined in an assembly that is not referenced

In a build of the Pri-1 tests:

https://dev.azure.com/dnceng/public/_build/results?buildId=1640719&view=logs&j=db768a9e-3325-5ce3-2d52-896c40bee68a&t=59112ad9-fe58-56ee-6943-84df6f0e3f8e

```
    eventsourceerror -> /__w/1/s/artifacts/tests/coreclr/AnyOS.x64.Checked/tracing/eventpipe/eventsourceerror/eventsourceerror/eventsourceerror.dll
/__w/1/s/src/tests/tracing/eventpipe/eventsvalidation/ExceptionThrown_V1.cs(24,20): error CS0012: The type 'EventPipeProvider' is defined in an assembly that is not referenced. You must add a reference to assembly 'Microsoft.Diagnostics.NETCore.Client, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null'. [/__w/1/s/src/tests/tracing/eventpipe/eventsvalidation/ExceptionThrown_V1.csproj] [/__w/1/s/src/tests/build.proj]
##[error]src/tests/tracing/eventpipe/eventsvalidation/ExceptionThrown_V1.cs(24,20): error CS0012: The type 'EventPipeProvider' is defined in an assembly that is not referenced. You must add a reference to assembly 'Microsoft.Diagnostics.NETCore.Client, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null'. [/__w/1/s/src/tests/tracing/eventpipe/eventsvalidation/ExceptionThrown_V1.csproj]
/__w/1/s/src/tests/tracing/eventpipe/eventsvalidation/GCEvents.cs(25,20): error CS0012: The type 'EventPipeProvider' is defined in an assembly that is not referenced. You must add a reference to assembly 'Microsoft.Diagnostics.NETCore.Client, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null'. [/__w/1/s/src/tests/tracing/eventpipe/eventsvalidation/GCEvents.csproj] [/__w/1/s/src/tests/build.proj]
##[error]src/tests/tracing/eventpipe/eventsvalidation/GCEvents.cs(25,20): error CS0012: The type 'EventPipeProvider' is defined in an assembly that is not referenced. You must add a reference to assembly 'Microsoft.Diagnostics.NETCore.Client, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null'. [/__w/1/s/src/tests/tracing/eventpipe/eventsvalidation/GCEvents.csproj]
/__w/1/s/src/tests/tracing/eventpipe/eventsvalidation/GCFinalizers.cs(25,20): error CS0012: The type 'EventPipeProvider' is defined in an assembly that is not referenced. You must add a reference to assembly 'Microsoft.Diagnostics.NETCore.Client, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null'. [/__w/1/s/src/tests/tracing/eventpipe/eventsvalidation/GCFinalizers.csproj] [/__w/1/s/src/tests/build.proj]
##[error]src/tests/tracing/eventpipe/eventsvalidation/GCFinalizers.cs(25,20): error CS0012: The type 'EventPipeProvider' is defined in an assembly that is not referenced. You must add a reference to assembly 'Microsoft.Diagnostics.NETCore.Client, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null'. [/__w/1/s/src/tests/tracing/eventpipe/eventsvalidation/GCFinalizers.csproj]
```

A rolling outerloop job will presumably hit this soon:

https://dev.azure.com/dnceng/public/_build/results?buildId=1640881&view=logs&j=db768a9e-3325-5ce3-2d52-896c40bee68a

It seems likely this is related to https://github.com/dotnet/runtime/pull/64358

@mdh1418 @hoyosjs

--- a/src/tests/tracing/eventpipe/eventsvalidation/ExceptionThrown_V1.cs
+++ b/src/tests/tracing/eventpipe/eventsvalidation/ExceptionThrown_V1.cs
@@ -4,8 +4,8 @@
 using System;
 using System.Diagnostics.Tracing;
 using System.Collections.Generic;
-using Microsoft.Diagnostics.Tools.RuntimeClient;
 using Tracing.Tests.Common;
+using Microsoft.Diagnostics.NETCore.Client;
 
 namespace Tracing.Tests.ExceptionThrown_V1
 {
@@ -13,15 +13,14 @@ namespace Tracing.Tests.ExceptionThrown_V1
     {
         public static int Main(string[] args)
         {
-            var providers = new List<Provider>()
+            var providers = new List<EventPipeProvider>()
             {
-                new Provider("Microsoft-DotNETCore-SampleProfiler"),
+                new EventPipeProvider("Microsoft-DotNETCore-SampleProfiler", EventLevel.Verbose),
                 //ExceptionKeyword (0x8000): 0b1000_0000_0000_0000
-                new Provider("Microsoft-Windows-DotNETRuntime", 0b1000_0000_0000_0000, EventLevel.Warning)
+                new EventPipeProvider("Microsoft-Windows-DotNETRuntime", 0b1000_0000_0000_0000, EventLevel.Warning)
             };
 
-            var configuration = new SessionConfiguration(circularBufferSizeMB: 1024, format: EventPipeSerializationFormat.NetTrace,  providers: providers);
-            return IpcTraceTest.RunAndValidateEventCounts(_expectedEventCounts, _eventGeneratingAction, configuration);
+            return IpcTraceTest.RunAndValidateEventCounts(_expectedEventCounts, _eventGeneratingAction, providers, 1024);
         }
 
         private static Dictionary<string, ExpectedEventCount> _expectedEventCounts = new Dictionary<string, ExpectedEventCount>()
@@ -31,7 +30,7 @@ namespace Tracing.Tests.ExceptionThrown_V1
             { "Microsoft-DotNETCore-SampleProfiler", -1 }
         };
 
-        private static Action _eventGeneratingAction = () => 
+        private static Action _eventGeneratingAction = () =>
         {
             for (int i = 0; i < 1000; i++)
             {
@@ -40,7 +39,7 @@ namespace Tracing.Tests.ExceptionThrown_V1
                 try
                 {
                     throw new ArgumentNullException("Throw ArgumentNullException");
-                } 
+                }
                 catch (Exception e)
                 {
                     //Do nothing

--- a/src/tests/tracing/eventpipe/eventsvalidation/ExceptionThrown_V1.cs
+++ b/src/tests/tracing/eventpipe/eventsvalidation/ExceptionThrown_V1.cs
@@ -17,7 +17,7 @@ namespace Tracing.Tests.ExceptionThrown_V1
             {
                 new EventPipeProvider("Microsoft-DotNETCore-SampleProfiler", EventLevel.Verbose),
                 //ExceptionKeyword (0x8000): 0b1000_0000_0000_0000
-                new EventPipeProvider("Microsoft-Windows-DotNETRuntime", 0b1000_0000_0000_0000, EventLevel.Warning)
+                new EventPipeProvider("Microsoft-Windows-DotNETRuntime", EventLevel.Warning, 0b1000_0000_0000_0000)
             };
 
             return IpcTraceTest.RunAndValidateEventCounts(_expectedEventCounts, _eventGeneratingAction, providers, 1024);

--- a/src/tests/tracing/eventpipe/eventsvalidation/ExceptionThrown_V1.csproj
+++ b/src/tests/tracing/eventpipe/eventsvalidation/ExceptionThrown_V1.csproj
@@ -14,5 +14,6 @@
   <ItemGroup>
     <Compile Include="ExceptionThrown_V1.cs" />
     <ProjectReference Include="../common/common.csproj" />
+    <ProjectReference Include="../common/Microsoft.Diagnostics.NETCore.Client/Microsoft.Diagnostics.NETCore.Client.csproj" />
   </ItemGroup>
 </Project>

--- a/src/tests/tracing/eventpipe/eventsvalidation/GCEvents.cs
+++ b/src/tests/tracing/eventpipe/eventsvalidation/GCEvents.cs
@@ -16,9 +16,9 @@ namespace Tracing.Tests.GCEvents
         {
             var providers = new List<EventPipeProvider>()
             {
-                new EventPipeProvider("Microsoft-DotNETCore-SampleProfiler"),
+                new EventPipeProvider("Microsoft-DotNETCore-SampleProfiler", EventLevel.Verbose),
                 //GCKeyword (0x1): 0b1
-                new EventPipeProvider("Microsoft-Windows-DotNETRuntime", 0b1, EventLevel.Informational)
+                new EventPipeProvider("Microsoft-Windows-DotNETRuntime", EventLevel.Informational, 0b1)
             };
 
             return IpcTraceTest.RunAndValidateEventCounts(_expectedEventCounts, _eventGeneratingAction, providers, 1024, _DoesTraceContainEvents);

--- a/src/tests/tracing/eventpipe/eventsvalidation/GCEvents.cs
+++ b/src/tests/tracing/eventpipe/eventsvalidation/GCEvents.cs
@@ -4,9 +4,9 @@
 using System;
 using System.Diagnostics.Tracing;
 using System.Collections.Generic;
-using Microsoft.Diagnostics.Tools.RuntimeClient;
 using Microsoft.Diagnostics.Tracing;
 using Tracing.Tests.Common;
+using Microsoft.Diagnostics.NETCore.Client;
 
 namespace Tracing.Tests.GCEvents
 {
@@ -14,15 +14,14 @@ namespace Tracing.Tests.GCEvents
     {
         public static int Main(string[] args)
         {
-            var providers = new List<Provider>()
+            var providers = new List<EventPipeProvider>()
             {
-                new Provider("Microsoft-DotNETCore-SampleProfiler"),
+                new EventPipeProvider("Microsoft-DotNETCore-SampleProfiler"),
                 //GCKeyword (0x1): 0b1
-                new Provider("Microsoft-Windows-DotNETRuntime", 0b1, EventLevel.Informational)
+                new EventPipeProvider("Microsoft-Windows-DotNETRuntime", 0b1, EventLevel.Informational)
             };
-            
-            var configuration = new SessionConfiguration(circularBufferSizeMB: 1024, format: EventPipeSerializationFormat.NetTrace,  providers: providers);
-            return IpcTraceTest.RunAndValidateEventCounts(_expectedEventCounts, _eventGeneratingAction, configuration, _DoesTraceContainEvents);
+
+            return IpcTraceTest.RunAndValidateEventCounts(_expectedEventCounts, _eventGeneratingAction, providers, 1024, _DoesTraceContainEvents);
         }
 
         private static Dictionary<string, ExpectedEventCount> _expectedEventCounts = new Dictionary<string, ExpectedEventCount>()
@@ -32,7 +31,7 @@ namespace Tracing.Tests.GCEvents
             { "Microsoft-DotNETCore-SampleProfiler", -1 }
         };
 
-        private static Action _eventGeneratingAction = () => 
+        private static Action _eventGeneratingAction = () =>
         {
             for (int i = 0; i < 50; i++)
             {
@@ -44,7 +43,7 @@ namespace Tracing.Tests.GCEvents
             }
         };
 
-        private static Func<EventPipeEventSource, Func<int>> _DoesTraceContainEvents = (source) => 
+        private static Func<EventPipeEventSource, Func<int>> _DoesTraceContainEvents = (source) =>
         {
             int GCStartEvents = 0;
             int GCEndEvents = 0;
@@ -52,9 +51,9 @@ namespace Tracing.Tests.GCEvents
             source.Clr.GCStop += (eventData) => GCEndEvents += 1;
 
             int GCRestartEEStartEvents = 0;
-            int GCRestartEEStopEvents = 0;           
+            int GCRestartEEStopEvents = 0;
             source.Clr.GCRestartEEStart += (eventData) => GCRestartEEStartEvents += 1;
-            source.Clr.GCRestartEEStop += (eventData) => GCRestartEEStopEvents += 1; 
+            source.Clr.GCRestartEEStop += (eventData) => GCRestartEEStopEvents += 1;
 
             int GCSuspendEEEvents = 0;
             int GCSuspendEEEndEvents = 0;

--- a/src/tests/tracing/eventpipe/eventsvalidation/GCEvents.csproj
+++ b/src/tests/tracing/eventpipe/eventsvalidation/GCEvents.csproj
@@ -11,5 +11,6 @@
   <ItemGroup>
     <Compile Include="GCEvents.cs" />
     <ProjectReference Include="../common/common.csproj" />
+    <ProjectReference Include="../common/Microsoft.Diagnostics.NETCore.Client/Microsoft.Diagnostics.NETCore.Client.csproj" />
   </ItemGroup>
 </Project>

--- a/src/tests/tracing/eventpipe/eventsvalidation/GCFinalizers.cs
+++ b/src/tests/tracing/eventpipe/eventsvalidation/GCFinalizers.cs
@@ -4,7 +4,7 @@
 using System;
 using System.Diagnostics.Tracing;
 using System.Collections.Generic;
-using Microsoft.Diagnostics.Tools.RuntimeClient;
+using Microsoft.Diagnostics.NETCore.Client;
 using Microsoft.Diagnostics.Tracing;
 using Tracing.Tests.Common;
 
@@ -14,15 +14,14 @@ namespace Tracing.Tests.GCFinalizers
     {
         public static int Main(string[] args)
         {
-            var providers = new List<Provider>()
+            var providers = new List<EventPipeProvider>()
             {
-                new Provider("Microsoft-DotNETCore-SampleProfiler"),
+                new EventPipeProvider("Microsoft-DotNETCore-SampleProfiler"),
                 //GCKeyword (0x1): 0b1
-                new Provider("Microsoft-Windows-DotNETRuntime", 0b1, EventLevel.Informational)
+                new EventPipeProvider("Microsoft-Windows-DotNETRuntime", 0b1, EventLevel.Informational)
             };
-            
-            var configuration = new SessionConfiguration(circularBufferSizeMB: 1024, format: EventPipeSerializationFormat.NetTrace,  providers: providers);
-            return IpcTraceTest.RunAndValidateEventCounts(_expectedEventCounts, _eventGeneratingAction, configuration, _DoesTraceContainEvents);
+
+            return IpcTraceTest.RunAndValidateEventCounts(_expectedEventCounts, _eventGeneratingAction, providers, 1024, _DoesTraceContainEvents);
         }
 
         private static Dictionary<string, ExpectedEventCount> _expectedEventCounts = new Dictionary<string, ExpectedEventCount>()
@@ -32,7 +31,7 @@ namespace Tracing.Tests.GCFinalizers
             { "Microsoft-DotNETCore-SampleProfiler", -1 }
         };
 
-        private static Action _eventGeneratingAction = () => 
+        private static Action _eventGeneratingAction = () =>
         {
             for (int i = 0; i < 50; i++)
             {
@@ -47,7 +46,7 @@ namespace Tracing.Tests.GCFinalizers
             GC.Collect();
         };
 
-        private static Func<EventPipeEventSource, Func<int>> _DoesTraceContainEvents = (source) => 
+        private static Func<EventPipeEventSource, Func<int>> _DoesTraceContainEvents = (source) =>
         {
             int GCFinalizersEndEvents = 0;
             source.Clr.GCFinalizersStop += (eventData) => GCFinalizersEndEvents += 1;

--- a/src/tests/tracing/eventpipe/eventsvalidation/GCFinalizers.cs
+++ b/src/tests/tracing/eventpipe/eventsvalidation/GCFinalizers.cs
@@ -16,9 +16,9 @@ namespace Tracing.Tests.GCFinalizers
         {
             var providers = new List<EventPipeProvider>()
             {
-                new EventPipeProvider("Microsoft-DotNETCore-SampleProfiler"),
+                new EventPipeProvider("Microsoft-DotNETCore-SampleProfiler", EventLevel.Verbose),
                 //GCKeyword (0x1): 0b1
-                new EventPipeProvider("Microsoft-Windows-DotNETRuntime", 0b1, EventLevel.Informational)
+                new EventPipeProvider("Microsoft-Windows-DotNETRuntime", EventLevel.Informational, 0b1)
             };
 
             return IpcTraceTest.RunAndValidateEventCounts(_expectedEventCounts, _eventGeneratingAction, providers, 1024, _DoesTraceContainEvents);

--- a/src/tests/tracing/eventpipe/eventsvalidation/GCFinalizers.csproj
+++ b/src/tests/tracing/eventpipe/eventsvalidation/GCFinalizers.csproj
@@ -11,5 +11,6 @@
   <ItemGroup>
     <Compile Include="GCFinalizers.cs" />
     <ProjectReference Include="../common/common.csproj" />
+    <ProjectReference Include="../common/Microsoft.Diagnostics.NETCore.Client/Microsoft.Diagnostics.NETCore.Client.csproj" />
   </ItemGroup>
 </Project>

